src/templateDialog.js

```diff
@@ -0,0 +1,214 @@
+import DetailContainer from './DetailContainer'
+import ErrorAlert from './errorAlert'
+import {
+  updateEditTemplateName,
+  updateEditTemplateDescription,
+  updateEditTemplateOS,
+  updateEditTemplateMemory,
+  updateEditTemplateCpu,
+  changeEditTemplate,
+  editTemplate,
+  closeDetail,
+} from './actions'
+
+class templateDialog extends React.Component {
+  constructor (props) {
+    super(props)
+    this.changeTemplate = this.changeTemplate.bind(this)
+    this.changeOperatingSystem = this.changeOperatingSystem.bind(this)
+    this.changeMemory = this.changeMemory.bind(this)
+    this.changeCpu = this.changeCpu.bind(this)
+    this.changeName = this.changeName.bind(this)
+    this.changeDescription = this.changeDescription.bind(this)
+    this.editTemplate = this.editTemplate.bind(this)
+    this.closeDialog = this.closeDialog.bind(this)
+    this.getTemplate = this.getTemplate.bind(this)
+    this.getOperatingSystem = this.getOperatingSystem.bind(this)
+  }
+
+  componentDidUpdate () {
+    // without this template combobox its not rerendering
+    $(this.templateOs).selectpicker('refresh')
+  }
+
+  componentDidMount () {
+    $(this.template).selectpicker()
+    $(this.templateOs).selectpicker()
+  }
+
+  changeName () {
+    this.props.changeName(this.templateName.value)
+  }
+
+  getOperatingSystem (name) {
+    return this.props.operatingSystems.get('operatingSystems').toList().find(os =>
+      os.get('description') === name)
+  }
+
+  getTemplate (name) {
+    return this.props.templates.get('templates').toList().find(template =>
+      template.get('name') === name)
+  }
+
+  changeTemplate () {
+    const template = this.getTemplate(this.template.value)
+    this.props.changeTemplate(template)
+  }
+
+  changeOperatingSystem () {
+    this.props.changeOperatingSystem(this.templateOs.value)
+  }
+
+  changeDescription () {
+    this.props.changeDescription(this.templateDescription.value)
+  }
+
+  changeMemory () {
+    this.props.changeMemory(this.templateMemory.value)
+  }
+
+  changeCpu () {
+    this.props.changeCpu(this.templateCpus.value)
+  }
+
+  closeDialog (e) {
+    e.preventDefault()
+    this.props.closeDialog()
+  }
+
+  editTemplate (e) {
+    e.preventDefault()
+    const template = {
+      'name': this.templateName.value,
+      'description': this.templateDescription.value,
+      'memory': this.templateMemory.value,
+      'cpu': {
+        'topology': {
+          'cores': '1',
+          'sockets': this.templateCpus.value,
+          'threads': '1',
+        },
+      },
+      'os': {
+        'type': this.templateOs.value,
+      },
+    }
+    this.props.editTemplate(template, this.props.template.get('id'))
+  }
+
+  render () {
+    return (
+      <DetailContainer>
+        <h1>Edit Template</h1>
+        <hr />
+        <ErrorAlert message={this.props.errorMessage} />
+        <form className='form-horizontal'>
+          <LabeledSelect
+            label='Templates'
```

Comment:
feel free to ping @vojtechszocs and @sjd78 for their thoughts on i18n, especially if you're going to hook into Zanata like ovirt-engine-dashboard does.